src/pip/_internal/req/constructors.py

```diff
@@ -71,11 +71,24 @@ def parse_editable(editable_req):
         - a requirement name
         - an URL
         - extras
-        - editable options
     Accepted requirements:
-        svn+http://blahblah@rev#egg=Foobar[baz]&subdirectory=version_subdir
-        .[some_extra]
+        - svn+http://blahblah@rev#egg=Foobar[baz]&subdirectory=version_subdir
+        - local_path[some_extra]
+        - Foobar[extra] @ svn+http://blahblah@rev#subdirectory=subdir ; markers
     """
+    try:
+        req = Requirement(editable_req)
+    except InvalidRequirement:
+        pass
+    else:
+        if req.url:
+            # Join the marker back into the name part. This will be parsed out
+            # later into a Requirement again.
+            if req.marker:
+                name = f"{req.name} ; {req.marker}"
+            else:
+                name = req.name
+            return (name, req.url, req.extras)
```

Comment:
Hm, on second look... this will let non VCS URLs go through ? So we need to merge https://github.com/pypa/pip/pull/9436 first and see how we can do the `link.is_vcs` here too.

Reply:
Oh, good point, `Requirement()` does not check if the URL is actually valid.

Reply:
One thing we ca do here is to add an extra check, and only treat the string as PEP 508 if the URL part contains `://`. This means pseudo URL strings will automatically fall back to the old parsing logic, and can be handled independently to this new logic.

Reply:
@sbidoul I added an additional check to prevent pseudo-URLs from going through here. Do you think it’d help?

Reply:
@uranusjr I have not tested (I whish I had more time for pip...) but this version would accept any URLs ? Editables URLs must be `file://` or `vcs+...://` urls only, correct ? And even file URLs must point to a local directory, not a file.

Reply:
Correct. There are already checks in the later stage to error out on those values, so I think it’s fine to pass them through here.

https://github.com/pypa/pip/blob/baaf66f70d84cf3367d5b43fec16762ca9703810/src/pip/_internal/req/constructors.py#L116-L119